programs/server/play.html

```diff
@@ -517,11 +517,12 @@
     let previous_query = '';
 
     const current_url = new URL(window.location);
+    const opened_locally = document.location.href.startsWith("file://");
```

Comment:
Is location.protocol ok?

Reply:
I'm not a JS expert but I believe checking href is same as checking location protocol.